src/acl/NoteData.cc

```diff
@@ -49,6 +49,18 @@ ACLNoteData::parse()
 {
     char* t = ConfigParser::strtokFile();
     assert (t != nullptr);
+
+    if (!name.isEmpty() && name.cmp(t) != 0) {
+        debugs(28, DBG_CRITICAL, "ERROR: Ignoring conflicting 'note' ACL configuration:" <<
+               Debug::Extra << "honored annotation name: " << name <<
+               Debug::Extra << "ignored annotation name: " << t <<
+               Debug::Extra << "configuration location: " << ConfigParser::CurrentLocation() <<
+               Debug::Extra << "advice: To match annotations with different names, " <<
+               "use note ACLs with different names " <<
+               "(that may be ORed using an 'any-of' ACL).");
+        return;
+    }
+
     name = t;
```

Comment:
IMO, configurations with misconfigured key-changing note, req_header, or rep_header ACLs should be rejected, but I think that change should be done (for all three ACLs at once) in a dedicated PR. This PR is still useful as a step that unifies the relevant note, req_header, or rep_header ACL behavior without changing req_header and rep_header.

However, if others insist, I can close this PR and post a PR that rejects the corresponding malformed configurations (for all three ACLs).

Reply:
I agree with this being a CRITICAL level error, it has security implications on many installations.
By that same security reason critical configuration errors should not continue to let Squid run. I do insist on that.

This issue should be handled the same way as if the admin changed the type of an already named ACL.
Which is currently the old destruct mechanism (https://github.com/squid-cache/squid/blob/master/src/acl/Acl.cc#L235).

Reply:
> Alex: if others insist, I can close this PR and post a PR that rejects the corresponding malformed configurations (for all three ACLs).

> Amos: I do insist on that.

Closing. Will post a configuration-rejecting PR.